File: 10324 Zeros and Ones/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		int cases = 0;
		String s;
		StringBuilder output = new StringBuilder();
		while((s = br.readLine()) != null) {
		    output.append("Case ").append(++cases).append(":\n");
		    int[] groups = new int[s.length()];
		    for(int i = 1; i < s.length(); ++i)
                groups[i] = groups[i - 1] + (s.charAt(i) != s.charAt(i - 1) ? 1 : 0);
            st.nextToken();
            int n = (int)st.nval;
            while(n-- > 0) {
                st.nextToken();
                int i = (int)st.nval;
                st.nextToken();
                int j = (int)st.nval;
                output.append(groups[i] == groups[j] ? "Yes\n" : "No\n");
            }
		}
		System.out.print(output);
	}
}

File: 10324 Zeros and Ones/solution.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    int cases = 0;
    char *buffer = NULL;
    while(scanf("%ms", &buffer) == 1) {
        printf("Case %d:\n", ++cases);
        int len = strlen(buffer), groups[len];
        for(int i = 1; i < len; ++i)
            groups[i] = groups[i - 1] + (buffer[i] != buffer[i - 1]);
        int n;
        scanf("%d", &n);
        while(n--) {
            int i, j;
            scanf("%d %d", &i, &j);
            puts(groups[i] == groups[j] ? "Yes" : "No");
        }
        free(buffer);
        buffer = NULL;
    }
    return 0;
}
